src/overlays/actors/ovl_En_Ge2/z_en_ge2.c

```diff
@@ -39,78 +35,738 @@ const ActorInit En_Ge2_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8B5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8B6B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8B7A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8B848.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8B90C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BA40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BB3C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BC1C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BC78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BCEC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BD38.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BD90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BE08.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BF04.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8BFC8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C048.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C0B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C13C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C45C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C59C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C644.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8C9B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/EnGe2_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8CC0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8CCB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/func_80B8CCFC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge2/EnGe2_Draw.s")
+void EnGe2_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnGe2* this = (EnGe2*)thisx;
+
+    ActorShape_Init(&this->picto.actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gGerudoPurpleSkel, NULL, this->jointTable, this->morphTable,
+                       GERUDO_PURPLE_LIMB_MAX);
+    Animation_PlayLoop(&this->skelAnime, &gGerudoPurpleWalkingAnim);
+
+    Collider_InitAndSetCylinder(play, &this->collider, &this->picto.actor, &sCylinderInit);
+    this->picto.actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->picto.actor, 0.01f);
+    this->picto.actor.uncullZoneForward = 1200.0f;
+
+    if (this->picto.actor.world.rot.z == 0) {
+        this->verticalDetectRange = 40.0f;
+    } else {
+        this->verticalDetectRange = fabsf(this->picto.actor.world.rot.z * 20.0f);
+    }
```

Comment:
Usually we don't for simple params, although this case is a bit weird. It might also be better for it to be visible that it's using rot for this? The other problem is that it's the range / 20, which is not very easy to describe succinctly.

Reply:
```c
    if (GERUDO_PURPLE_GET_VERTICAL_RANGE(&this->picto.actor) == 0) {
        this->verticalDetectRange = 40.0f;
    } else {
        this->verticalDetectRange = fabsf(GERUDO_PURPLE_GET_VERTICAL_RANGE(&this->picto.actor) * 20.0f);
    }
```
It would look like this kind of thing, not sure it's actually better since this is the only place it's used.

Reply:
I would still just call it _range_, yeah its multiplied by some scalar but that's likely just in case they want the range to be longer and worry about the rotation as a parameter being limited to max [0-360], 360 units of range is... tiny.